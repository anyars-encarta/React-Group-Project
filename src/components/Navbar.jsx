import { NavLink } from 'react-router-dom';
import '../styles/navbar.css';
import logo from '../assets/planet-grey.png';

const Navbar = () => (
  <header>
    <div className="logo-container">
      <img className="logo-img" alt="logo" src={logo} />
      <p>Space Travelers&apos; Hub</p>
    </div>
    <nav className="navbar">
      <ul className="nav-items">
        <li><NavLink to="/" className={({ isActive }) => (isActive ? 'active link' : 'pending link')}>Rockets</NavLink></li>
        <li><NavLink to="/missions" className="link">Missions</NavLink></li>
        <li>|</li>
        <li><NavLink to="/my-profile" className="link">My Profile</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Navbar;
