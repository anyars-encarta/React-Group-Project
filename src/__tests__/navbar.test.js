import React from 'react';
import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('renders without errors', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    expect(screen.getByText("Space Travelers' Hub")).toBeInTheDocument();
  });

  const tree = renderer.create(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('navigates to Rockets page when Rockets link is clicked', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );

  const rocketsNavLink = screen.getByText('Rockets');
  expect(rocketsNavLink).toBeInTheDocument();

  act(() => {
    userEvent.click(rocketsNavLink);
  });

  expect(window.location.pathname).toBe('/');
});

it('navigates to Missions page when Missions link is clicked', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );

  const missionsNavLink = screen.getByText('Missions');
  expect(missionsNavLink).toBeInTheDocument();

  act(() => {
    userEvent.click(missionsNavLink);
  });

  expect(window.location.pathname).toBe('/missions');
});
