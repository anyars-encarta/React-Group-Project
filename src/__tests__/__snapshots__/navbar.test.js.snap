// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<header>
  <div
    className="nav-container"
  >
    <div
      className="logo-container"
    >
      <img
        alt="logo"
        className="logo-img"
        src="planet-color-2.png"
      />
      <p>
        Space Travelers' Hub
      </p>
    </div>
    <nav
      className="navbar"
    >
      <ul
        className="nav-items"
      >
        <li>
          <a
            aria-current="page"
            className="active link"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className="link"
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li>
          |
        </li>
        <li>
          <a
            className="link"
            href="/my-profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
